homeassistant/components/philips_js/media_player.py

```diff
@@ -63,18 +71,38 @@ def __init__(self, tv, name, on_script):
         """Initialize the Philips TV."""
         self._tv = tv
         self._name = name
-        self._state = None
-        self._volume = None
-        self._muted = False
-        self._program_name = None
-        self._channel_name = None
-        self._source = None
-        self._source_list = []
-        self._connfail = 0
-        self._source_mapping = {}
-        self._watching_tv = None
-        self._channel_name = None
+        self._sources = {}
+        self._channels = {}
         self._on_script = on_script
+        self._supports = SUPPORT_PHILIPS_JS
+        if self._on_script:
+            self._supports |= SUPPORT_TURN_ON
+        self._update_task = None
+
+    def _update_soon(self, delay):
+        """Reschedule update task."""
+        if self._update_task:
+            self._update_task()
+            self._update_task = None
+
+        self.schedule_update_ha_state(
+            force_refresh=False)
+
+        def update_forced(event_time):
+            self.schedule_update_ha_state(force_refresh=True)
+
+        def update_and_restart(event_time):
+            update_forced(event_time)
+            self._update_task = track_time_interval(
+                self.hass, update_forced,
+                timedelta(seconds=DEFAULT_SCAN_INTERVAL))
+
+        call_later(self.hass, delay, update_and_restart)
+
+    async def async_added_to_hass(self):
+        """Start running updates once we are added to hass."""
+        self.hass.add_job(
```

Comment:
you cannot use `add_job` inside an async function. You would want to `await self.hass.async_add_executor_job(self._update_soon)`

Reply:
Also, if you want to trigger an update on add, you should just pass `True` as second argument to `add_entities`

Reply:
That block setup. if TV is not turned on, it can take a quite some time for the network connection to time out.